megatron/learning_rates.py

```diff
@@ -102,6 +105,9 @@ def get_lr(self):
             coeff = (1.0 - decay_ratio)
         elif self.decay_style == 'cosine':
             coeff = 0.5 * (math.cos(math.pi * decay_ratio) + 1.0)
+        elif self.decay_style == 'inverse_sqrt':
+            return self.max_lr * math.sqrt(float(max(self.warmup_steps, 1))) / \
+                math.sqrt(float(max(num_steps_,1)))
```

Comment:
Can you write in math formulas what you want here? I don't think it does the same thing as T5 for example in this case. (it should scale in `rsqrt(num_steps)` instead of `sqrt(warmup_step/num_steps)`). Your implementation seems to correspond to https://cs.github.com/google-research/t5x/blob/730967370cab41b4ce51419a40538fe359e07d7b/t5x/utils.py#L793 If you want the original one you should do the following.

```suggestion
            return self.max_lr / math.sqrt(float(max(num_steps_, 1)))
```

Some things to note:
 -  the `min_lr` is going to trigger before, typically if `self.num_token > self.decay_tokens`. So you'll see a big change in `lr` then. Typically I think you might want to move this code before the `if self.decay_tokens` since I don't think you want to support min_lr (otherwise you need to figure out hot to make you lr schedule continuous.
  - I think you should ask around which decay they use for PaLM for example. I'm guessing that it's `rsqrt_decay` according to the paper, but I'm not sure. Either way probably a good thing to support all of them.
   - I'm guessing your using `warmup_steps` the same as `offsets`

Reply:
Also wondering if you actually need to cast to float before?

Reply:
We probably dont need to cast as `math.sqrt` return float either way